python/cuml/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -828,3 +855,35 @@ class RandomForestClassifier(BaseRandomForestModel,
         if self.dtype == np.float64:
             return get_rf_json(rf_forest64).decode('utf-8')
         return get_rf_json(rf_forest).decode('utf-8')
+
+    def cpu_to_gpu(self):
+        # treelite does an internal isinstance check to detect an sklearn
+        # RF, which proxymodule interferes with. We work around that
+        # temporarily here just for treelite internal check and
+        # restore the __class__ at the end of the method.
+        if GlobalSettings().accelerator_active:
+            cls_cahed = self._cpu_model.__class__
+            self._cpu_model.__class__ = sys.modules['sklearn.ensemble'].RandomForestClassifier
```

Comment:
Let's put a lock around this manipulation to avoid messing things up in a multithreaded context.